rectangle-calculations/main.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import "fmt"
+
+type Rectangle struct {
+	length float32
+	width  float32
+}
+
+func (rectangle Rectangle) calculateArea() float32 {
+	return rectangle.length * rectangle.width
+}
+
+func (rectangle Rectangle) calculatePerimeter() float32 {
+	return 2 * (rectangle.length + rectangle.width)
+}
+
+func main() {
+	var rectangle Rectangle
+
+	fmt.Printf("\nPlease enter length of rectangle\n")
+	fmt.Scanln(&rectangle.length)
+
+	fmt.Printf("\nPlease enter width of rectangle\n")
+	fmt.Scanln(&rectangle.width)
+
+	if rectangle.length < 1 || rectangle.width < 1 {
```

Comment:
Since the question doesn't specify that the numbers can't be zero, we only need to check that the length and breadth are non-negative (they can be zero)